shell/components/firebase-upload.js

```diff
@@ -0,0 +1,62 @@
+import Xen from './xen/xen.js';
+import Firebase from './firebase.js';
+
+const {firebase, storage} = Firebase;
+
+const template = `
+  <style>
+    ::slotted(*) {
+      cursor: pointer;
+    }
+    input[type="file"] {
+      display: none;
+    }
+  </style>
+  <label>
+    <slot></slot>
+    <input type="file" accept="{{accept}}" multiple="{{multiple}}" on-change="_onFilesChanged">
+  </label>
+`;
+
+class FirebaseUpload extends Xen.Base {
+  get template() {
+    return template;
+  }
+  static get observedAttributes() {
+    return ['multiple', 'accept'];
+  }
+  _render({accept, multiple}) {
+    return {
+      accept,
+      multiple
+    };
+  }
+  _onFilesChanged(e) {
+    const input = e.currentTarget;
+    const file = input.files[0];
```

Comment:
We allow optionally selecting multiple but currently we only upload the first -- add a TODO comment or throw NotImplemented or something more hard core?

Reply:
Perhaps we get this dumb version checked in, and then you can do other PRs to salt to taste? My intent was sort of just to get this ball rolling.

Reply:
Sure, yeah, that was my intent re: hitting approve ahead of time. I wasn't sure if you'd want to do it. I can address my comments in another patch.